packages/expo-sqlite/src/next/Database.ts

```diff
@@ -0,0 +1,177 @@
+import { EventEmitter, Subscription } from 'expo-modules-core';
+
+import ExpoSQLite from './ExpoSQLiteNext';
+import { NativeDatabase, OpenOptions } from './NativeDatabase';
+import { BindParams, RunResult, Statement, VariadicBindParams } from './Statement';
+
+export { OpenOptions };
+
+const emitter = new EventEmitter(ExpoSQLite);
+
+/**
+ * A SQLite database.
+ */
+export class Database {
+  constructor(private readonly nativeDatabase: NativeDatabase) {}
+
+  /**
+   * Synchronous call to return whether the database is currently in a transaction.
+   */
+  public isInTransaction(): boolean {
+    return this.nativeDatabase.isInTransaction();
+  }
+
+  /**
+   * Asynchronous call to return whether the database is currently in a transaction.
+   */
+  public isInTransactionAsync(): Promise<boolean> {
+    return this.nativeDatabase.isInTransactionAsync();
+  }
+
+  /**
+   * Close the database.
+   */
+  public closeAsync(): Promise<void> {
+    return this.nativeDatabase.closeAsync();
+  }
+
+  /**
+   * Execute all SQL queries in the supplied string.
+   * > Note: The queries are not escaped for you! Be careful when constructing your queries.
+   *
+   * @param source A string containing all the SQL queries.
+   */
+  public execAsync(source: string): Promise<void> {
+    return this.nativeDatabase.execAsync(source);
+  }
+
+  /**
+   * Prepare a SQL statement.
+   *
+   * @param source A string containing the SQL query.
+   * @returns A `Statement` object.
+   */
+  public async prepareAsync(source: string): Promise<Statement> {
+    const nativeStatement = new ExpoSQLite.NativeStatement();
+    await this.nativeDatabase.prepareAsync(nativeStatement, source);
+    return new Statement(this.nativeDatabase, nativeStatement);
+  }
+
+  /**
+   * Execute a transaction and automatically commit/rollback based on the `txn` success.
+   *
+   * @param txn An async function to execute within a transaction.
+   */
+  public async transactionAsync(txn: () => Promise<void>): Promise<void> {
+    try {
+      await this.nativeDatabase.execAsync('BEGIN');
```

Comment:
this seems susceptible to the problem that matt brought up in notion: https://www.notion.so/CRSqlite-REST-syncing-example-69b0d4423d5547c98f24b80fe2323071?d=1f7423c8e5ce4510b26abfcac9bc0815&pvs=4#172d47cb5f794c90aeb4f69f70af4b2b

in general, what do you think about adding support for sync functions to the api as well?

Reply:
i have been convinced by Brent on a call to support sync apis. my only concern is about [the api naming](https://linear.app/expo/issue/ENG-10388/having-synchronous-api-from-expo-sqlitenext). i will do in a separate pr.

we also talked about the problem of async transaction api. i'm still looking for a proper solution.

Reply:
> this seems susceptible to the problem that matt brought up in notion: https://www.notion.so/CRSqlite-REST-syncing-example-69b0d4423d5547c98f24b80fe2323071?d=1f7423c8e5ce4510b26abfcac9bc0815&pvs=4#172d47cb5f794c90aeb4f69f70af4b2b 

address by https://github.com/expo/expo/pull/24812/commits/c11839e0ddab23a7606758be1f7ad3738f58e0f2 from the new `transactionExclusiveAsync` call. close ENG-10394